### PR TITLE
[WIP] sqlalchemy_utils for ColorType, JSONType, PasswordType, ScalarListType

### PR DESCRIPTION
**Pull request by colin99d**
This PR attempts to add the remaining sqlachemy_utils data types. Complete list here: https://sqlalchemy-utils.readthedocs.io/en/latest/data_types.html#module-sqlalchemy_utils.types.choice. Notes for each one appear below:

- ArrowType: already implemented 
- **ChoiceType**: got this to work, but implementation is not super user friendly. Would love suggestions for a better implementation
- CompositeType: postgres specific and I use mysql (:, skipping
- **ColorType**: implemented
- CountryType, CurrencyType, LocalType, WeekDaysType: currently breaks if nothing is sent to the form, this seems to be common among all forms using Babel. @aminalaee do you have any suggestion on how I could fix this?
- : already implemented
- EmailType: Did in #150
- EncryptedType: currently this uses a binary encryption which is not practical to type into a form. Based on their [documentation](https://sqlalchemy-utils.readthedocs.io/en/latest/_modules/sqlalchemy_utils/types/encrypted/encrypted_type.html#EncryptedType) this will be changed to a string implementation in the future and we can implement then.
- **JSONType**: implemented
- LtreeType: postgres specific and I use mysql (:, skipping
- IPAddressType: Did in #150
- **PasswordType**: implemented
- PhoneNumberType: This uses composite type, how do we want to implement it?
- **ScalarListType**: implemented, however adding an item does not work well (I still think a bad implementation is better than the presence of the field crashing the server)
- TimezoneType: implemented
- TSVectorType: postgres specific and I use mysql (:, skipping
- URLType: already implemented
- UUIDType: already implemented

### CODE REVIEW

**Comment by aminalaee** [comment]
Any updates?

File: sqladmin/forms.py [hunk 1]
```diff
@@ -364,6 +365,22 @@ def conv_ip_address(
         kwargs["validators"].append(validators.IPAddress(ipv4=True, ipv6=True))
         return StringField(**kwargs)
 
+    @converts("sqlalchemy_utils.types.choice.ChoiceType")
+    def conv_choice(
+        self, model: type, prop: ColumnProperty, kwargs: Dict[str, Any]
+    ) -> UnboundField:
```
Comment (aminalaee) [inline sqladmin/forms.py:371]: https://github.com/flask-admin/flask-admin/blob/7cff9c742d44d42a8d3495c73a6d71381c796396/flask_admin/contrib/sqla/form.py#L301

File: scripts/install [hunk 2]
```diff
@@ -14,4 +14,6 @@ else
 fi
 
 "$PIP" install -r "$REQUIREMENTS"
+"$PIP" install colour
+"$PIP" install passlib
```
[the remark this answers is not in the document]
Comment (aminalaee) [reply scripts/install:18]: yeah, those are specified in `setup.cfg` file

**Comment by colin99d** [comment]
Hey man, I did my best to implement choices like you suggested but ran into errors. Any way I could get some help?

**Comment by aminalaee** [comment]
@colin99d Thanks, I think the pipeline is failing for the `black` formatting, but if you see it's a blocker, just leave it outside the PR so we can pick it up later.

And also please rename the PR so it's clear what types are added.

**Comment by colin99d** [comment]
So I should just exclude choices for now?

**Comment by aminalaee** [comment]
Yes please.

**Comment by aminalaee** [comment]
Are you still interested in updating this?

**Comment by colin99d** [comment]
Sorry kinda spaced it! Do you just need me to get CI to pass?

**Pull request closed without merging**